docs/release-notes.md

```diff
@@ -25,6 +25,7 @@ nav_order: 9
 - Fix reproducibility of systemd preset file in ignition-apply output
 - Clarify spec docs for `files`/`directories`/`links` `group` fields
 - Document that `user`/`group` fields aren't applied to hard links
+- Fail if files/links/dirs conflict with systemd units or dropins
```

Comment:
This isn't really a feature but it isn't really a bugfix either.  I'd put it in Changes.